serving/src/main/java/feast/serving/controller/ServingServiceGRpcController.java

```diff
@@ -70,6 +77,8 @@ public void getFeastServingInfo(
   public void getOnlineFeatures(
       GetOnlineFeaturesRequest request,
       StreamObserver<GetOnlineFeaturesResponse> responseObserver) {
+    this.authorizationService.authorizeRequest(
```

Comment:
### Multiple Projects
Online Serving (`getOnlineFeatures()`) support 2 cases for supplying projects:
- Project is specified in the request. In this case only one project can be retrieved per request. This code handles that case properly
- Project is not specified in the request. Leading `getOnlineFeatures()` to look at FeatureReferences in the request for the project. Since multiple unique projects can be referenced in the Feature references , and we need to ensure that the user is authorized for all unique projects specified in the request.

### Latency
This is probably a duplicate of @woop comment on latency.
Online Serving is latency sensitive, Perhaps we would to cache authorization requests to Core for a duration of time. (ie authorization grant is valid for a duration of time before having to re verify with Core). See `CachedSpecService` for a similar cache implementation for reference

Reply:
From what I understood from the code, if project is not specified on the api `client.get_online_features() or client.get_historical_features()` a default value set in the config will be passed. I see that requested features could be in different feature sets but I don't see an option to refer to multiple projects in a single request. Correct me if my understanding is wrong. Below is what the pydoc says, `feature_refs: List of feature references that will be returned for each entity.
                Each feature reference should have the following format:
                "feature_set:feature" where "feature_set" & "feature" refer to
                the feature and feature set names respectively.
                Only the feature name is required.`

Reply:
I definitely don't want to break existing clients but feast is 0.x, so we should weigh the tradeoff of more complexity vs a breaking change.

Reply:
Caching is definitely desirable but we just made the move to externalize authorization decisions to another service.  There could also be caching there but I think given the nature of online serving, it makes sense to also cache the results from that service in serving directly.  It would have to be a pretty short and small cache though I would think.  Hopefully the number of users x projects is small for online serving.  In our case it should be.

Reply:
@mrzzy I see what you mean, I had a look at the v0.4 sdk. Caching is definitely one solution, How about we propose an adaption for the clients fetching features from multiple projects. The adoption would be to add access to `default` project for all such clients. I see that client passes `project=default` if not set. That being said, irrespective of we propose adoption or not, caching would be helpful.

Reply:
> The adoption would be to add access to default project for all such clients. I see that client passes project=default if not set.

This seems to be a hacky way of solving things. I don't think we want a situation where we have to deal with special project `default` with special properties.

> There could also be caching there but I think given the nature of online serving, it makes sense to also cache the results from that service in serving directly.

Yeah, I am agreement on this one. Feast Serving already caches the Feature Set specs it pulls from Feast Core in [`CachedSpecService`](https://github.com/feast-dev/feast/blob/master/serving/src/main/java/feast/serving/specs/CachedSpecService.java). We could have something similar for Feast Serving.

Reply:
Ok so if I can summarize, we will have a cache at the `authorizeReques` level?

Reply:
Yes, that is correct.

Reply:
@mrzzy Have a look at the updated implementation and let me know it makes sense.